src/pspm_interpolate.m

```diff
@@ -65,238 +51,169 @@
 end
 outdata = {}; % will return a cell of the same size as the indata
 sts = -1;
-switch length(varargin)
-  case 1
-    indata = varargin{1};
-    options = struct();
-  case 2
-    indata = varargin{1};
-    options = varargin{2};
-  case 3
-    warning('Up to two variables are accepted by pspm_interpolate.');
-    return
-end
+
 % 1.1 check input arguments
 if nargin<1
-  warning('ID:missing_data', 'No data.\n');
+  warning('ID:missing_data', 'Don''t know what to do.\n');
   return;
-end
-if isempty(indata)
+elseif isempty(indata)
   warning('ID:missing_data', 'Input data is empty, nothing to do.');
   return;
 end
+
+if isnumeric(indata) 
+    if nargin >= 2
+        options = varargin{1};
```

Comment:
There is no issue for this situation, but I have a feeling that this refers to indata=numeric_array and `channel` does not exist in this case, so `nargin` can be only 1 or 2. ">2" is a situation that is unexpected.